### PR TITLE
[bug]add escape character option to global dictionary

### PR DESCRIPTION
**Pull request by eason-lyx**
while using spark-csv to load the data,by default,the escape char is "\".
so when the data in quote with "\",then spark-csv will skip \,like "ab\c"->"abc".
and in csv reader ,it will no skip.

so there is inconsistency in read in phase 1 - dictionary generation and phase 2- data loading.

then we provide an option to user to specify escape character.
@Vimal-Das please review.thanks.

### CODE REVIEW

**Comment by QiangCai** [comment]
LGTM

**Comment by eason-lyx** [comment]
retest this please

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestLoadDataWithHiveSyntax.scala [hunk 1]
```diff
@@ -201,10 +201,29 @@ class TestLoadDataWithHiveSyntax extends QueryTest with BeforeAndAfterAll {
...
+    sql("""
+           CREATE TABLE IF NOT EXISTS t3
+           (ID Int, date Timestamp, country String,
+           name String, phonetype String, serialname String, salary Int)
+           STORED BY 'org.apache.carbondata.format'
+           """)
+
+    sql(s"""
+           LOAD DATA LOCAL INPATH './src/test/resources/datawithbackslash.csv' into table t3
+           OPTIONS('ESCAPECHAR'='@')
```
Comment (QiangCai) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataload/TestLoadDataWithHiveSyntax.scala]: why use '@' ,but not  '"'

**Comment by QiangCai** [comment]
OK,  merge to master.

**Pull request merged**